### PR TITLE
DRAFT development: nixification of dendrite-pinecone and element-packet-forwarder apps

### PR DESCRIPTION
**Pull request by enesoztrk**
## Checklist for things done

- [ ] Summary of the proposed changes in the PR description
- [ ] More detailed description in the commit message(s)
- [ ] Commits are squashed into relevant entities - avoid a lot of minimal dev time commits in the PR
- [ ] [Contribution guidelines](https://github.com/tiiuae/ghaf/blob/main/CONTRIBUTING.md) followed
- [ ] Ghaf documentation updated with the commit - https://tiiuae.github.io/ghaf/
- [ ] PR linked to architecture documentation and requirement(s) (ticket id)
- [ ] Test procedure described (or includes tests). Select one or more:
  - [ ] Tested on Lenovo X1 `x86_64`
  - [ ] Tested on Jetson Orin NX or AGX `aarch64`
  - [ ] Tested on Polarfire `riscv64`
- [ ] Author has run `nix flake check --accept-flake-config` and it passes
- [ ] All automatic Github Action checks pass - see [actions](https://github.com/tiiuae/ghaf/actions)
- [ ] Author has added reviewers and removed PR draft status

### CODE REVIEW

**Review by vilvo — CHANGES_REQUESTED** [review]
Thanks for making the approach visible early.

Overlays are not needed for:
- rust - [this](https://github.com/tiiuae/ghaf/pull/425/files#diff-b381aefa347d72a0a7754a252c986ac330d304d0375f79b34dd4e9255a3dc0baR16-R20) should stay within element-packet-forwarder packaging because we are not planning to bring (and overlay) rust development environment to ghaf.
- element-packet-forwarder - [instead of overlaying unnecessarily, just include this package directly](https://github.com/tiiuae/ghaf/pull/425/files#diff-b46313d81071f90bf9325f748e082010dd2acf5e669f30369c814c1d7521608e)

Also, I see some issues with:
- element-desktop patching - most seems just some dev time hacks (e.g. darwin and win32) so it should be properly cleaned only for what's required
- dendrite-pinecone patching - turnserver requires documentation for the context
- don't add to env.systemPackages but ghaf user packages instead - https://github.com/tiiuae/ghaf/pull/425/files#diff-51b61fcdf71428d0b03132a2564ed9123924cdfd86012256dd391a55e384b5b2R41 - or rather a separate unprivileged user to run the packet forwarder daemon.

**Comment by vilvo** [comment]
For the next review request:
- mark PR draft (not only DRAFT in the PR topic)
- please fix the issues in prev comment
- add PR description (even when draft)
- mark as many checklist items done as possible (even when draft)
- add testing instructions (so we can involve testing people)

File: targets/lenovo-x1-carbon.nix [hunk 1]
```diff
@@ -307,9 +312,53 @@
...
+                    name = "element";
+                    packages = [pkgs.element-desktop pkgs.dendrite-pinecone];
+                    macAddress = "02:00:00:03:08:01";
+                    ramMb = 3072;
+                    cores = 4;
+                    
+                      extraModules = [
+                      {
+                        systemd.services."dendrite-pinecone" = {
+                          description = "Dendrite is a second-generation Matrix homeserver with Pinecone which a next-generation P2P overlay network";
```
Comment (mikatammi) [inline targets/lenovo-x1-carbon.nix:325]: Should this be "which is a" ?

Comment (enesoztrk) [reply targets/lenovo-x1-carbon.nix]: thanks, it has been resolved.

**Comment by Mic92** [comment]
Just as a heads-up in case you are not closely following the matrix ecosystem:

Dendrite is mainly developed by Element and currently not under active development because
Element wants to become more focused (i.e. only working on Synapse) until they are in a financial more stable position.

**Pull request closed without merging**